simplemonitor/Monitors/gmirror.py

```diff
@@ -0,0 +1,56 @@
+"""
+Gmirror array checks for simplemonitor.
+"""
+
+import subprocess
+
+from .monitor import Monitor, register
+
+
+@register
+class MonitorGmirrorStatus(Monitor):
+    """
+    Check gmirror status for specified device.
+    """
+
+    monitor_type = "gmirror_status"
+
+    def __init__(self, name: str, config_options: dict) -> None:
+        super().__init__(name, config_options)
+        self.array_device = self.get_config_option(
+            "array_device", required_type="str", required=True
+        )
+        self.expected_disks = self.get_config_option(
+            "expected_disks", required_type="int", required=True
+        )
+
+    def run_test(self) -> bool:
+        """
+        Run `gmirror status` for the specified device. Keep the logic simpler
+        by requiring each device and the # of expected disks to be specified
+        separately.
+        """
+
+        run_cmd = ["gmirror", "status", "-gs", self.array_device]
+        try:
+            result = subprocess.run(run_cmd, capture_output=True, check=True)
+        except subprocess.CalledProcessError:
+            return self.record_fail("gmirror command failed")
+
+        status_lines = result.stdout.decode("utf-8").rstrip("\n").split("\n")
+
+        # Status should be same for the array, so just grab first line
+        status = status_lines[0].split()[1]
+
+        # Infer # of disks based on the number of lines
+        disk_count = len(status_lines)
+
+        msg = f"Array is in state {status} with {disk_count} disks"
+
+        if status == "COMPLETE" and disk_count == 2:
```

Comment:
Should this be `expected_disks`?

Reply:
Hi - yes, I'd been going back and forth on the naming... thanks for pointing this out... I'll fix the inconsistencies.

Reply:
Any other thoughts / feedback? I can convert this from a draft, but if you've got some ideas on testing, might make sense to try to add some tests first?

Reply:
Sorry, been a busy couple of weeks! Previously for this kind of thing I put a script in `tests/mocks/` which printed fake output equivalent to the real command either succeeding or failing, but I then started learning more about actually mocking libraries :)

If you have a look in `tests/test_service.py` the first set of tests show mocking return values and stdout etc for the `service` command, so that would probably been best for this too if you're happy doing that. (I'm hoping the way I'm doing it is suitable, I've been feeling my way through setting that stuff up so no idea if I'm doing it the best way :)

Reply:
Thanks; that's very helpful. Will try to take a look soon.

In the meantime, I took a look at (hopefully) fixing some timezone related stuff I observed with unrelated tests in #1598